### ISSUE TITLE
Content leader and content authorship

### ISSUE THREAD

**Issue by DopiePanda** [jasonrohrer/AnotherPlanetData#4]
I was asked by quite a few players if we could transfer over most of, or all current objects from OHOL. This is of course a community debate wether this gets done or not. But that is not my issue with it. My issue with it is that nothing except the rocket is credited to you with authorship.

This means that if some future content leader does decide to move over all OHOL items, the person who exports all of the items will be credited for creating almost 5k objects, 3k+ sprites, 5k animations, etc. **This means that the rest of the community needs to create close to 15.000 credited files to share half of the community part of the reward, while the other half would go to a single person who did nothing but export, add and commit.**

Another side note is that transitions are not included in mod exports, so the content leader needs to get these separately send outside of the exported mod, or have to create them for all of the mods recieved by the players. Is this intended?

**Comment by connorhsm** [comment]
@jasonrohrer 

Related to above https://github.com/jasonrohrer/OneLife/issues/1057

In addition to this, could you please provide clarification regarding profit sharing?
I assume that naturally there will be many more transitions than sprites and a sprite may be used multiple times, will different types have different weights?

**Comment by jasonrohrer** [comment]
As far as I'm aware, exporting objects that have no author hash will PRESERVE the lack of author hash on import.

This is true of all the OHOL objects (like the people) that I already exported and imported.  I just tested it again, and yes, importing my OHOL stuff keeps the black authorship hash.

So I think that problem is solved...

I mean, unless a malevolent content leader goes in and TOUCHES each imported object in the editor to add a hash.  They could also do this to "steal" authorship from other contributors.  Or they could just do a search-replace in the text files.

If they do this, they are a bad Content Leader, and they should be voted out, and then their changes should be rolled back in the github repo.

Since it's all public, people can keep an eye on this kind of thing, if they need to.


Aside from the 60%, I haven't worked out what the content share will be.

Sprites will be measured by their size, I think.  Bigger sprites generally take more work than small ones.

Objects will be measured by their complexity in terms of number of moving parts, and animations can also be measured in terms of how.... complex they are.  Like how many parameters are not 0.  Making a simple wheel spin is easy.  Animating a walking animation for an octopus is hard.

Sounds... well, I'm not sure how those will be counted.  Probably just by the number of sounds, with each sound being worth the same.

Music and BG textures have no authorship tracking, but we will have to track that manually somehow.

And transitions will be counted too, which is where the Content Leader---even if they are just doing the work of gluing stuff together and testing it---will get credit.

I haven't taken the time to think it through, but I think that Sprites and Animations are more work than Objects.

So maybe something like:

35% for Sprites and background textures
20% for Animations
10% for Objects
10% for Sounds
15% for Transitions
10% for music

I.e., we take 35% of the money, and give it to the people who made Sprites and BG textures.  We divide that portion of the money up among the contributors based on the portion of the total pixels that each person contributed.

Obviously, for a lot of things, the same person will make the Sprites, Object, and Animations... they will have a contribution in each area.

But if you make an object by cobbling together someone else's sprites (or OHOL sprites), then you just get credit for the Object and Animaions.

**Comment by Arcurus** [comment]
Maybe a more fair approach would be to give the content leader some % of what content was added while he was the leader.

**Comment by jasonrohrer** [comment]
Yeah, there's no double-authorship tracking like that, currently.

But that's where counting the Transitions will come in.

**Comment by DopiePanda** [comment]
@jasonrohrer 

I adjusted the spawn rate of Bay Trees, and it re-wrote authorship to me. Is this supposed to happen?
![image](https://github.com/jasonrohrer/AnotherPlanetData/assets/494864/5f6ac649-0d29-40a0-a5e3-bc1d5543df80)

I would assume that no object with an authorship tag added could ever be updated, especially not automatically like this
